saya/core/src/prover/atlantic/client.rs

```diff
@@ -145,24 +251,54 @@ impl AtlanticClient {
 
         let response = self.http_client.get(url).send().await?;
         if !response.status().is_success() {
-            anyhow::bail!("unsuccessful status code: {}", response.status());
+            return Err(ProverError::Prover(format!(
+                "unsuccessful status code: {}",
+                response.status()
+            )));
         }
 
         let response = response.json::<AtlanticQueryJobsResponse>().await?;
         Ok(response.jobs)
     }
+    pub async fn get_atlantic_query(self, id: &str) -> Result<AtlanticQueryResponse, ProverError> {
+        let mut url = self.api_base.clone();
+        url.path_segments_mut()
+            .unwrap()
+            .push("atlantic-query")
+            .push(id);
+        let response = self.http_client.get(url).send().await?;
+        if !response.status().is_success() {
+            return Err(ProverError::Prover(format!(
+                "unsuccessful status code: {}",
+                response.status()
+            )));
+        }
+        let response = response.json::<AtlanticQueryResponse>().await?;
+        Ok(response)
+    }
 
-    pub async fn get_proof(&self, id: &str) -> Result<String> {
-        let url = format!(
-            "https://atlantic-queries.s3.nl-ams.scw.cloud/sharp_queries/query_{}/proof.json",
-            id
-        );
+    pub async fn get_proof(&self, id: &str) -> Result<String, ProverError> {
+        let url = format!("{}/{}/proof.json", ATLANTIC_S3_BASE, id);
 
         let response = self.http_client.get(url).send().await?;
         if !response.status().is_success() {
-            anyhow::bail!("unsuccessful status code: {}", response.status());
+            return Err(ProverError::Prover(format!(
+                "unsuccessful status code: {}\n{}",
+                response.status(),
+                response.text().await?
+            )));
         }
 
         Ok(response.text().await?)
     }
+
+    pub async fn get_trace(&self, id: &str) -> Result<Vec<u8>, ProverError> {
+        // For now we hardcode the cairo version to 0, since we only use pythonVM with Cairo 0 for layout bridge
+        // for the trace generation.
+        // But since the response from atlantic job doesn't contain the cairo version,
+        // we need to have some state to keep this information.
+        let url = format!("{}/{}/pie.cairo0.zip", ATLANTIC_S3_BASE, id);
```

Comment:
For this one (see comment right above), we may actually use the DB to keep this context.

But that's actually annoying to have to keep state just for the name of the file.